ceno_zkvm/src/instructions/riscv/rv32im/mmu.rs

```diff
@@ -142,14 +138,9 @@ impl MemPadder {
         new_len: usize,
         records: Vec<MemInitRecord>,
     ) -> Vec<MemInitRecord> {
-        if records.is_empty() {
-            self.padded(new_len, records)
-        } else {
-            self.padded(new_len, records)
-                .into_iter()
-                .sorted_by_key(|record| record.addr)
-                .collect()
-        }
+        let mut padded = self.padded(new_len, records);
+        padded.sort_by_key(|record| record.addr);
+        padded
```

Comment:
This does extra work without being clearer when we just want the default content. If you really want to remove that `if` then it's more meaningful to split it into two functions.

Reply:
Oh, I didn't actually care that much about the `if`.  It was more about converting from vector to iterator (then internally in `sorted_by_key` convert back to vector and sort then convert back to iterator) and then back to vector again.  I'm not at all against letting the computer do some extra work, but here it didn't even make the code easier to read.

But while I was at it, I also simplified the structured by removing the `if`.  The fewer branching code paths you have, the easier it is to achieve good test coverage of all the code paths.  Rust's sorting is very efficient for already sorted data: it finishes in O(n).  And we need O(n) anyway just to produce the data in the first place.

Reply:
See https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.sorted_by_key